docs/source/index.rst

```diff
@@ -103,3 +103,9 @@ Please see the `developer’s guide`_ for contributing and `communication`_ for
    contributor-guide/roadmap
    contributor-guide/quarterly_roadmap
    contributor-guide/specification/index
+
+.. toctree::
```

Comment:
I tried to hide this from the sidebar but it seems to show up:

![Screenshot 2024-04-22 at 8 35 23 PM](https://github.com/apache/datafusion/assets/490673/62a720ab-592b-42d3-9d72-ec541003c131)

Any suggestions appreciated